src/porepy/numerics/fv/biot.py

```diff
@@ -453,15 +400,131 @@ def build_rhs_normals_single_dimension(dim):
+        # as a force on the faces. The right hand side is thus formed of the
+        # unit vector.
+        def build_rhs_units_single_dimension(dim):
+            vals = np.ones(num_subfno_unique)
+            ind = subcell_topology.subfno_unique
+            mat = sps.coo_matrix(
+                (vals, (ind, ind)), shape=(num_subfno_unique, num_subfno_unique)
+            )
+            return mat
+
+        rhs_units = build_rhs_units_single_dimension(0)
+
+        for i in range(1, nd):
+            this_dim = build_rhs_units_single_dimension(i)
+            rhs_units = sps.block_diag([rhs_units, this_dim])
+
+        rhs_units = bound_exclusion_mech.exclude_dirichlet(rhs_units)
+
+        num_dir_subface = (
+            bound_exclusion_mech.exclude_neu.shape[1]
+            - bound_exclusion_mech.exclude_neu.shape[0]
+        )
+
+        # No right hand side for cell displacement equations.
+        rhs_units_displ_var = sps.coo_matrix(
+            (nd * num_subfno - num_dir_subface, num_subfno_unique * nd)
+        )
+
+        # Why minus?
+        rhs_units = -sps.vstack([rhs_units, rhs_units_displ_var])
+        del rhs_units_displ_var
+
+        # Mapping from sub-cell to cell-centre values
+        def build_sc2c_single_dimension(dim):
+            rows = np.arange(sub_cell_index[dim].size)
+            cols = cell_node_blocks[0]
+            vals = np.ones(rows.size)
+            mat = sps.coo_matrix(
+                (vals, (rows, cols)), shape=(sub_cell_index[dim].size, g.num_cells)
+            ).tocsr()
+            return mat
+
+        sc2c = build_sc2c_single_dimension(0)
+        for i in range(1, nd):
+            this_dim = build_sc2c_single_dimension(i)
+            sc2c = sps.vstack([sc2c, this_dim])
+
+        # computation of imbalance coefficients,
+        # that is jumps in cell-centers pressures
+        grad_p_jumps = hf2f * hook * igrad * rhs_units * unique_nAlpha_grad * sc2c
+
+        # mapping from subface to unique subface for multi-dim problems
+        vals = np.ones(num_subfno_unique * nd)
+        rows = fvutils.expand_indices_nd(subcell_topology.subfno_unique, nd)
+        cols = fvutils.expand_indices_incr(
+            subcell_topology.unique_subfno, nd, num_subhfno
+        )
+        map_unique_subfno = sps.coo_matrix(
+            (vals, (rows, cols)), shape=(num_subfno_unique * nd, num_subhfno * nd)
+        )
+
+        del vals, rows, cols
+
+        # Computation of the "hydrostatic" term,
+        # that is the actual force on the face
+        grad_p_force = hf2f * map_unique_subfno * nAlpha_grad * sc2c
+
+        matrices_m["grad_p_jumps"] = grad_p_jumps
+        matrices_m["grad_p_force"] = grad_p_force
+
+        stabilization = div * igrad * rhs_units * unique_nAlpha_grad * sc2c
+        # include in the stabilization term also the "hydrostatic term?
+        # I believe so but this needs some more testing
+        stabilization += div * nAlpha_grad * sc2c
```

Comment:
the "hydrostatic" term is grad_p_force.
The notation is by analogy with the new mpfa discretization with gravity.

Reply:
Please find a better name - the analogue is not clear, in particular since nobody outside our group has thought about mpfa and gravity in that particular way.